src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateNew = JSON.parse(JSON.stringify(state));
+
+  for (let key = 0; key < actions.length; ++key) {
+    if (actions[key].type === 'addProperties') {
+      const stateNew1 = Object.assign({}, stateNew, actions[key].extraData);
+
+      stateHistory.push(stateNew1);
+      stateNew = JSON.parse(JSON.stringify(stateNew1));
```

Comment:
[DONT REPEAT YOURSELF]: You are repeating the logic of pushing the new state to the `stateHistory` and cloning it afterwards in each case. This should be done just once after the `switch` block.